src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -699,7 +699,7 @@ private static SFLoginOutput newSession(
             sessionId,
             commonParams);
 
-    if (consentCacheIdToken) {
+    if (consentCacheIdToken && asBoolean(loginInput.getSessionParameters().get(CLIENT_STORE_TEMPORARY_CREDENTIAL))) {
```

Comment:
This is to make sure when users don't use SecureStorageService, that part of code won't be accessed.

Reply:
what about the case when they want to use but they don't have the dependency installed?

Reply:
Then errors pop up, which will break user's program.

Reply:
I wonder what exception will it throw and if it can be captured. Is it going to throw UnsatisifiedLinkError?

Reply:
Maybe no symbol found. Let me try it again.

Reply:
Something like this would happen
```
Exception in thread "main" java.lang.NoClassDefFoundError: com/sun/jna/Library
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:151)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:821)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:719)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:642)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:600)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at net.snowflake.client.core.SecureStorageAppleManager$SecurityLibManager.getInstance(SecureStorageAppleManager.java:193)
	at net.snowflake.client.core.SecureStorageAppleManager.<init>(SecureStorageAppleManager.java:21)
	at net.snowflake.client.core.SecureStorageAppleManager.builder(SecureStorageAppleManager.java:25)
	at net.snowflake.client.core.CredentialManager.<init>(CredentialManager.java:22)
	at net.snowflake.client.core.CredentialManager.getInstance(CredentialManager.java:36)
	at net.snowflake.client.core.SessionUtil.openSession(SessionUtil.java:263)
	at net.snowflake.client.core.SFSession.open(SFSession.java:596)
	at net.snowflake.client.jdbc.SnowflakeConnectionV1.initialize(SnowflakeConnectionV1.java:151)
	at net.snowflake.client.jdbc.SnowflakeConnectionV1.<init>(SnowflakeConnectionV1.java:112)
	at net.snowflake.client.jdbc.SnowflakeDriver.connect(SnowflakeDriver.java:142)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:189)
	at SnowflakeJDBCExample.getConnection(SnowflakeJDBCExample.java:117)
	at SnowflakeJDBCExample.main(SnowflakeJDBCExample.java:33)
Caused by: java.lang.ClassNotFoundException: com.sun.jna.Library
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 23 more

Reply:
So in this particular case wouldn't it make sense to catch the NoClassDefFoundError and log it instead of failing? I am thinking of replicating what we have for Python connector. CC @sfc-gh-stakeda @sfc-gh-mknister

Reply:
I agree to catch NoClassDefFoundError and raise an nice message message.

Reply:
Thanks for all advice! I think we need to decide where to go. Whether raise an error with nicer message, which breaks user's program or just log it and go back to not using secure storage device, without stopping the program? @sfc-gh-stakeda , @sfc-gh-hchaturvedi

Reply:
Better not stopping the applications and continuing without the cache.

Reply:
We need a clear documentation about this change anyway.

Reply:
I agree with Shige.. Let's keep parity between drivers and not drop the connection. Handle the error fall back to without idToken connection and let the user see pop up. Meanwhile the error should be logged with a proper and clear message.